src/_pytest/pathlib.py

```diff
@@ -311,7 +326,7 @@ def resolve_from_str(input, root):
         return root.joinpath(input)
 
 
-def fnmatch_ex(pattern, path):
+def fnmatch_ex(pattern: str, path) -> bool:
     """FNMatcher port from py.path.common which works with PurePath() instances.
```

Comment:
Also `path` here?

Reply:
There is a `path = PurePath(path)` below which means it can have various types. This feels like something to clean up, so I didn't touch it for now.